app/src/main/res/navigation/navigation.xml

```diff
@@ -43,9 +43,7 @@
             app:enterAnim="@anim/slide_in_right"
             app:exitAnim="@anim/slide_out_left"
             app:popEnterAnim="@anim/slide_in_left"
-            app:popExitAnim="@anim/slide_out_right"
-            app:popUpTo="@id/welcomeFragment"
-            app:popUpToInclusive="true" />
```

Comment:
In general, I believe a navigation action should either have a `destination` (where we want to _advance_ to) or a `popUpTo` (where we want to _go back_ to). You had both defined on a couple actions, and it looks like the `destination` took priority, so I could safely remove the unused `popUpTo` and its metadata in `popUpToInclusive`.